### PR TITLE
feat: refine the behaviour of potion effects

### PR DESCRIPTION
**Pull request by heyhey123-git**
#276

### CODE REVIEW

File: server/src/main/java/org/allaymc/server/entity/component/EntityDamageComponentImpl.java [hunk 1]
```diff
@@ -45,8 +45,17 @@ public class EntityDamageComponentImpl implements EntityDamageComponent {
 
     @Override
     public boolean attack(DamageContainer damage) {
-        if (!canBeAttacked(damage)) return false;
-        if (!checkAndUpdateCoolDown(damage)) return false;
+        // Fire resistance effect
+        if (
```
Comment (smartcmd) [inline server/src/main/java/org/allaymc/server/entity/component/EntityDamageComponentImpl.java:49]: why don't move this check to canBeAttacked()?

**Review by smartcmd — CHANGES_REQUESTED** [review]

File: server/src/main/java/org/allaymc/server/entity/component/EntityBaseComponentImpl.java [hunk 2]
```diff
@@ -711,6 +711,11 @@ public void onFall() {
         this.fallDistance = 0;
     }
 
+    @Override
+    public Map<EffectType, EffectInstance> getAllEffects() {
+        return new HashMap<>(effects);
```
Comment (smartcmd) [inline server/src/main/java/org/allaymc/server/entity/component/EntityBaseComponentImpl.java:716]: Use UnmodificalMap here and add @UnmodificalView annotation to interface

**Review by smartcmd — APPROVED** [review]

**Review by smartcmd — CHANGES_REQUESTED** [review]
qwq

**Comment by heyhey123-git** [comment]
> qwq

qwq

**Review by smartcmd — APPROVED** [review]

**Pull request merged**